torchvision/datasets/kinetics.py

```diff
@@ -35,29 +48,92 @@ class Kinetics400(VisionDataset):
+        num_workers (int): Use multiple workers for VideoClips creation
+        _num_download_workers (int): Use multiprocessing in order to speed up download.
 
     Returns:
         tuple: A 3-tuple with the following entries:
 
-            - video (Tensor[T, H, W, C]): the `T` video frames
+            - video (Tensor[T, H, W, C]): the `T` video frames in torch.uint8 tensor
             - audio(Tensor[K, L]): the audio frames, where `K` is the number of channels
-              and `L` is the number of points
+              and `L` is the number of points in torch.float tensor
             - label (int): class of the video clip
+    
+    Raises:
+        RuntimeError: If ``download is True`` and the image archive is already extracted.
     """
 
-    def __init__(self, root, frames_per_clip, step_between_clips=1, frame_rate=None,
-                 extensions=('avi',), transform=None, _precomputed_metadata=None,
-                 num_workers=1, _video_width=0, _video_height=0,
-                 _video_min_dimension=0, _audio_samples=0, _audio_channels=0):
-        super(Kinetics400, self).__init__(root)
+    _FILES = {
+        400: "https://s3.amazonaws.com/kinetics/400/{split}/k400_{split}_path.txt",
+        600: "https://s3.amazonaws.com/kinetics/600/{split}/k600_{split}_path.txt",
+        700: "https://s3.amazonaws.com/kinetics/700_2020/{split}/k700_2020_{split}_path.txt",
+    }
+    _ANNOTATION = {
+        400: "https://s3.amazonaws.com/kinetics/400/annotations/{split}.csv",
+        600: "https://s3.amazonaws.com/kinetics/600/annotations/{split}.txt",
+        700: "https://s3.amazonaws.com/kinetics/700_2020/annotations/{split}.csv",
+    }
+
+    def __init__(
+        self,
+        root: str,
+        n_classes: int = 400,
+        split: str = "train",
+        frame_rate: float = None,
+        frames_per_clip: int = 5,
+        step_between_clips: int = 1,
+        annotation_path: str = None,
+        transform: Optional[Callable] = None,
+        extensions=("avi", "mp4"),
+        download: bool = False,
+        num_workers: int = 1,
+        _precomputed_metadata=None,
+        _num_download_workers=1,
+        _video_width=0,
+        _video_height=0,
+        _video_min_dimension=0,
+        _audio_samples=0,
+        _audio_channels=0,
+    ) -> None:
+
+        # TODO: support test
+        assert split in ["train", "val"]
+        assert n_classes in [400, 700]
+        self.n_classes = n_classes
+        self.extensions = extensions
+        self._num_download_workers = _num_download_workers
+
+        # set up self.root and self.split
+        self._set_up_paths(root, split)
 
+        # load annotation files
+        if annotation_path is not None:
+            self.annotations = pd.read_csv(annotation_path)
+        else:
+            self.annotations = pd.read_csv(
+                self._ANNOTATION[self.n_classes].format(split=self.split)
+            )
```

Comment:
Re https://github.com/pytorch/vision/pull/3680#issuecomment-821678875: The annotation files are structured like this:

```
label,youtube_id,time_start,time_end,split
clay pottery making,---0dWlqevI,19,29,train
...
```

It seems you are only using the fields `label` and `youtube_id`.

1. Is that correct? Shouldn't we respect `time_start` and `time_end`?
2. If we really only need two fields per row, we probably go the easy way:
    
    ```python
    import csv
    
    with open("annotations.csv") as fh:
        label, youtube_id = zip(*[row[:2] for row in csv.reader(fh, delimiter=",")])
    ```

    In case we also need access to the other fields, a [`DictReader`]() would probably a better choice. 

    In any case this will most likely be slower than the `pandas` implementation. We just recently removed the last `pandas` dependency in #3656. Before we reintroduce it here, we should profile this and see if it is a bottleneck.

Reply:
In theory yes, but the videos provided are already cut to the respective lens so we don't need them.

I'll rewrite this using `csv` then to avoid adding the additional dependencies.

Reply:
> In theory yes, but the videos provided are already cut to the respective lens so we don't need them.

Ah ok, so this is just the information how the file was obtained not instructions how to obtain it?

Reply:
I'll rewrite this with CSV then;
we don't need `time_start` and `time_end` as the official videos are already pre-trimmed.